src/trunk.c

```diff
@@ -4831,41 +5226,42 @@ trunk_compact_bundle(void *arg, void *scratch_buf)
    }
 
    /*
-    * 10. Clean up
+    * 9. Clean up
     */
    trunk_compact_bundle_cleanup_iterators(
       spl, &merge_itor, num_branches, skip_itor_arr);
 
    deinit_saved_pivots_in_scratch(scratch);
 
    /*
-    * 11. Reacquire read lock
-    */
-   trunk_node_get(spl->cc, req->addr, &node);
-
-   /*
-    * 12. For each newly split sibling replace bundle with new branch
+    * 11. For each newly split sibling replace bundle with new branch
     */
-   uint64 addr             = req->addr;
    uint64 num_replacements = 0;
-   uint64 generation;
-   do {
-      trunk_node_claim(spl->cc, &node);
-      trunk_node_lock(spl->cc, &node);
-
+   bool   should_continue  = TRUE;
+   while (should_continue) {
```

Comment:
This is basically a `key_compare`.  As above, I think the code will be easier to understand if the predicate is directly in the while condition.